thinc/types.py

```diff
@@ -1,19 +1,41 @@
+import sys
 from dataclasses import dataclass
+from enum import Enum
 from typing import (
-    Union,
-    Tuple,
+    TYPE_CHECKING,
+    Any,
     Callable,
-    Iterator,
-    Sized,
     Container,
-    Any,
-    Optional,
-    List,
     Generic,
+    Iterator,
+    List,
+    Optional,
+    Sized,
+    Tuple,
     TypeVar,
+    Union,
 )
-from enum import Enum
+
 import numpy
+from wasabi import msg
+
+# Literal shim, thanks Pydantic!
+# From: https://github.com/samuelcolvin/pydantic/blob/v1.3/pydantic/typing.py#L51-L60
+if sys.version_info < (3, 8):
+    if TYPE_CHECKING:
+        from typing_extensions import Literal
+    else:  # due to different mypy warnings raised during CI for python 3.7 and 3.8
+        try:
+            from typing_extensions import Literal
+        except ImportError:
+            msg.fail(
+                title="Incompatible System",
+                text="Either Python 3.8 must be used or the "
+                "`typing_extensions` module must be installed",
+                exits=1,
+            )
+else:
+    from typing import Literal
```

Comment:
If we do this, we should just make `typing_extensions` a dependency for < 3.8 (just like we currently do with `dataclasses` for < 3.7). And then we can assume that the user always has `Literal`.

Reply:
Yeah, I did that and forgot to remove it here. It's cleaned up now